litestar/openapi/plugins.py

```diff
@@ -0,0 +1,691 @@
+            path: Path to serve the OpenAPI UI at.
+            **kwargs: Additional arguments to pass to the base class.
+        """
+        self.js_url = js_url or f"https://cdn.jsdelivr.net/npm/@scalar/api-reference@{version}"
+        self.css_url = css_url or self._default_scalar_css_url
+        super().__init__(path=path, **kwargs)
+
+    def render(self, request: Request, openapi_schema: dict[str, Any]) -> bytes:
+        """Render an HTMl page for Scalar.
+
+        .. note:: Override this method to customize the template.
+
+        Args:
+            request: The request.
+            openapi_schema: The OpenAPI schema as a dictionary.
+
+        Returns:
+            A rendered html string.
+        """
+        head = f"""
+                  <head>
+                    <title>{openapi_schema["info"]["title"]}</title>
+                    <style>{self.style}</style>
+                    <meta charset="utf-8"/>
+                    <meta name="viewport" content="width=device-width, initial-scale=1">
+                    {self.favicon}
+                    <link rel="stylesheet" type="text/css" href="{self.css_url}">
+                  </head>
+                """
+
+        body = f"""
+                <noscript>
+                    Scalar requires Javascript to function. Please enable it to browse the documentation.
+                </noscript>
+                <script
+                  id="api-reference"
+                  data-url="openapi.json">
+                </script>
+                <script src="{self.js_url}" crossorigin></script>
+                """
+
+        return f"""
+                <!DOCTYPE html>
+                    <html>
+                        {head}
+                        {body}
+                    </html>
+                """.encode()
+
+    def receive_router(self, router: Router) -> None:
+        """Receive the router that serves the OpenAPI UI.
+
+        Adds a route to serve the Scalar.com OpenAPI UI CSS
+        pulled from the Litestar GitHub branding repo.
+
+        Args:
+            router: The router that serves the OpenAPI UI.
+        """
+        if self.css_url == self._default_scalar_css_url:
+            router.register(
+                get("/_litestar-internal/scalar.css", media_type=MediaType.CSS, sync_to_thread=False)(
+                    self.render_scalar_css
+                ),
+            )
+
+    def render_scalar_css(self) -> bytes:
+        """Fetch and render the Scalar CSS file from GitHub.
+
+        Returns:
+            The CSS content as bytes.
+        """
+        try:
+            with httpx.Client() as client:
+                response = client.get(self._default_litestar_css_url)
+                response.raise_for_status()
+                return response.content
+        except httpx.HTTPError:
+            here = Path(__file__).parent
+            css_path = here / "styles/scalar.css"
+            return css_path.read_bytes()
```

Comment:
Is all this necessary @JacobCoffee? We don't do anything like this for any of the other plugins to ensure that the css is available, and I can't see that the `openapi/styles/scalar.css` file exists.

Can we just leave it to hit the cdn for our style sheet like the others?

Reply:
was it b/c we were using a `githubusercontent` link before?

Reply:
@JacobCoffee I did this https://github.com/litestar-org/litestar/pull/2893/commits/6cdfe69ab7723b3745adbc6170c6ddf9855c7343 - but we can revert. Mainly I just wanted to see if it would get me across the line with coverage.